compiler/compiler/src/compiler.rs

```diff
@@ -67,11 +67,19 @@ impl<N: Network> Compiler<N> {
             source_file.absolute_start,
         )?;
 
-        // If the program is imported, then check that the name of its program scope matches the file name.
+        // Check that the name of its program scope matches the expected name.
         // Note that parsing enforces that there is exactly one program scope in a file.
-        // TODO: Clean up check.
         let program_scope = self.state.ast.ast.program_scopes.values().next().unwrap();
-        self.program_name = program_scope.program_id.name.to_string();
+        if self.program_name.is_empty() {
```

Comment:
Does it make more sense to `Option` this for the empty case in hopes of being more explicit?